doc/api/worker_threads.md

```diff
@@ -396,6 +396,74 @@ posting without having side effects.
 For more information on the serialization and deserialization mechanisms
 behind this API, see the [serialization API of the `v8` module][v8.serdes].
 
+#### Considerations when transferring TypedArrays and Buffers
+
+All `TypedArray` and `Buffer` instances are views over an underlying
+`ArrayBuffer`. That is, it is the `ArrayBuffer` that actually stores
+the raw data while the `TypedArray` and `Buffer` objects provide a
+way of viewing and manipulating the data. It is possible and common
+for multiple views to be created over the same `ArrayBuffer` instance.
+
+For example,
+
+```js
+const ab = new ArrayBuffer(10);
+
+// Both TypedArrays share the same underlying state.
+const u1 = new Uint8Array(ab);
+const u2 = new Uint16Array(ab);
+
+// Changing one, changes the other
+u1[1] = 1;
+
+console.log(u2[0]);  // prints 256
+```
```

Comment:
I think this example can be dropped, if we’re focusing on `MessagePort` here

Reply:
I'm ok with dropping it but originally included it to ensure that the reasons *why* transferring these things are unsafe. I could include this example back in the buffer.md docs but that would mean users would have to jump across multiple docs to get the full story.

Reply:
I mean, yeah, but if the idea is to warn about these things in multiple docs, then I’d prefer to document the shared bits in a single place, even if that means linking to multiple docs.

(Or just document it all in the buffer doc, possibly in various sections for the different use cases, and link to that from here.)